armi/materials/material.py

```diff
@@ -73,7 +87,8 @@ class Material(composites.Composite):
     with information about thermal scattering."""
 
     def __init__(self):
-        composites.Composite.__init__(self, self.__class__.name)
+        self.parent = None
```

Comment:
Why do materials still have parents? They shouldn't, since they're no longer in the hierarchical tree.

Remove or justify with implementation docs.

Reply:
It _appears_ to me that the mass fractions in all ARMI materials are designed to be changed by the user's choice of nuclides.  And right now those nuclide choices come from the parent.

Can you enlighten me on this?  Because these nuclides are also the reason materials have parameters.

ARMI materials seem to change a LOT based on the nuclides present in ARMI, and those nuclide are not static in ARMI.

Reply:
Materials used to be *the place* where composition (defined as number density vectors of elements and nuclides) was stored. We moved that to the `Component`'s `numberDensities` param long ago. 

However, we still initialize this param based on the `Material`s that users add to their problem via the `Material.setDefaultMassFracs` method, which sets `Material.p.massFrac`. This is indeed (and unfortunately) a bit coupled to nuclide choices, since sometimes users want to model *Carbon* while other times they want to model *Carbon-12*. 

So may be we need an intermediate ticket here, with the goal of: "Decouple composition definitions from neutronics library assumptions". 

Material definitions _should_ be static and evergreen, not dependent in any way on user modeling settings. This should include initial material composition, and probably also the process of building Component `numberDensities`, and potentially even the evolution of composition vs. time from transmutation and decay and reprocessing and everything else. The user modeling settings that map compositions to specific subsets of nuclides based on neutronics library needs should be isolated to the neutronics physics modules. For example, if a neutronics input is being built for a code that only has C-12 but the composition has Carbon, a mapping between carbon and C-12 should happen as the input is being written.

At this point, changing `setDefaultMassFracs` to `getDefaultMassFracs` in a way that just returns the data might be workable, and just eliminate `Material.p.massFrac` altogether. We'll have to see how much code accesses `p.massFrac` and assess how exactly to pull off an upgrade like this.

Reply:
Okay, I have made progress. I have removed all `Parameter`s from `Material` and `FuelMaterial` except:

* `massFrac`
* `puFrac`
* `uFrac`
* `zrFrac`

Frankly, those last three seem like a terrible idea to have in EVERY MATERIAL anyway.  

In the process of doing this cleaning I also fixed a couple of latent bugs in the materials, so that nice.

I'm hoping I'll remove these last 4 Paramters today, and we can re-open this PR.

Reply:
Done.

Okay, I removed a lot of code, so I'm sure the code coverage will go down. BUT `Material` not longer has any `Parameter`s.

Reply:
ok, please just add docs in the code explaining why there's a parent and what's to be done about it.